kustomize/resource_kustomization.go

```diff
@@ -330,7 +332,9 @@ func kustomizationResourceDelete(d *schema.ResourceData, m interface{}) error {
 
 	gvr, err := cgvk.getGVR(u.GroupVersionKind(), false)
 	if err != nil {
-		return fmt.Errorf("ResourceDelete: %s", err)
+		// If the Kind does not exist in the K8s API,
+		// the resource can't exist either
```

Comment:
I guess we don't need to worry about the kind of the resource to delete here, as the kind is from the the tf state which means it has to exist otherwise we couldn't have it already applied/imported into the state.

Reply:
Resources don't have dependencies. So TF creates and deletes them in arbitrary order. Meaning it's possible that the custom resource definition will be deleted before the custom resource. In that case, this API call that determines the URL of a resource, will fail, TF will error and the resource will stay int he state.

This change simply says, if we can't talk to the K8s API about the resource, there is no point considering it still existent in our state.

Reply:
Fair point.